Add default parameter selector to model fitting in muon analysis

Harrietbrown opened the pull request:
**Description of work.**

This PR adds methods to model fitting data selector so that it is able to by default plot log values or fit parameters.

**To test:**

1. Enable model analysis
1. Open muon analysis
1. Load MUSR 62260-5
1. Go to the fitting tab
1. Click simultaneous
1. Add any function
1. Go to the sequential tab
1. Fit all of the data
1. Go to the results tab
1. Select the sample_temp log
1. Create the results table
1. Go to the model analysis tab
1. The model analysis fit should be sample_temp on the x axis and the first parameter of the fit on the y axis.
1. return to the results tab and deselect sample_temp and select and sample_magn_field. Create the results table.
2. return to the model anaysis tab and now sample_magn_field should be on the x axis.

Fixes #32795

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

Harrietbrown committed:
Add default parameter selector to model analysis

This commit adds methods to model fitting data selector so that it is able to by default plot log values or fit parameters.

re: #32795
Change default parameter selection to work based on column type

re: #32795

AnthonyLim23 commented:
@Harrietbrown got some failing tests

AnthonyLim23 commented on qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/fitting_widgets/model_fitting/model_fitting_data_selector_view.py:
```diff
@@ -98,6 +100,22 @@ def set_selected_y_parameter(self, y_parameter: str) -> int:
+    def _get_x_parameter_for_update(self, old_x_parameter: str, x_parameters: list, x_parameter_types: list) -> str:
+        if self.y_selector.findText(old_x_parameter) != -1:
+            return old_x_parameter
+        elif 1 in x_parameter_types:
```
why 1 and 2?

Harrietbrown replied:
Column types in TableWorkspaces are enumerated so that X is 1, Y is 2. 
https://github.com/mantidproject/mantid/blob/c5e1ddb49cb0f49e8a4324af36bba9f266dbeb63/qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/results_tab_widget/results_tab_model.py#L24

AnthonyLim23 replied:
could we just put `x_col=1, y_col=2` at the top of the file and use them?

sf1919 commented:
I'm afraid this is crashing when I try to test it (specifically when I get to step 14 and try and generate a new results table). Is this the crash identified and fixed in #32938 ?

Harrietbrown committed:
change searching for axis type to use enumerated list

This commit changes the method of cheking the type of parameter in model fitting to use the same enum class that the results tab uses to build the table.

re: #32795

sf1919 reviewed (approved):
I have tested the PR and reviewed the code and (providing the fix from #32938 is implemented) this works well.
